src/main/java/io/github/thebusybiscuit/cscorelib2/inventory/InvUtils.java

```diff
@@ -31,6 +31,25 @@ public static boolean hasEmptySlot(@NonNull Inventory inv) {
         return inv.firstEmpty() != 1;
     }
 
+    /**
+     * This method checks both an ItemStack's and an Inventory's maxStackSize to determine
+     * if a given ItemStack can stack with another ItemStack in the given Inventory
+     * 
+     * @author kii-chan-reloaded
+     *
+     * @param stack
+     *            The ItemStack already in the inventory
+     * @param item
+     *            The ItemStack that shall be tested for
+     * @param inv
+     *            The Inventory these items are existing in
+     * @return Whether the maxStackSizes allow for these items to stack
+     */
+    private static boolean underMaxStackSize(@NonNull ItemStack stack, @NonNull ItemStack item, @NonNull Inventory inv) {
```

Comment:
As this is a private method, we don't need to perform the parameter validation here, in fact: These null checks have already been performed in the method below.
`lombok.NonNull` gets compiled into a preconditions null check.
I'd argue that for private method `javax.annotation.Nonnull` should be used instead.

Also suggested a name change to the method as `underMaxStackSize` is a bit non-descriptive.

```suggestion
    private static boolean isValidStackSize(@Nonnull ItemStack stack, @Nonnull ItemStack item, @Nonnull Inventory inv) {
```